helix-view/src/document.rs

```diff
@@ -1034,6 +1091,60 @@ impl Default for Document {
     }
 }
 
+#[derive(Clone, Copy, PartialEq, Debug, Default)]
+pub struct Config {
+    pub encoding: Option<&'static encoding::Encoding>,
+    pub line_ending: Option<LineEnding>,
+    pub indent_style: Option<IndentStyle>,
+    pub tab_width: Option<usize>,
+}
```

Comment:
These fields end up duplicated between Config and Document. Instead could we have a `config: Config` field on `Document` which is either derived from editorconfig or from language configuration and defaults that then gets used to hold these values in Document?

Reply:
Oh hmm on second though actually I do like the separate Config type. It is pretty elegant to determine the config and then hanlde that in `open_with_config`. Sorry for the flip-flopping 😅

On a more minor note, could we call this type `EditorConfig` instead of `Config`? I think I'm also a little caught up in the name like pickfire. We could clarify in a doc comment:

```rs
/// Configuration derived from EditorConfig. Despite the name, this configuration
/// applies to configuration that belongs to [`Document`] rather than `Editor`
/// like line-endings and encoding.
#[derive(..)]
pub struct EditorConfig {
  // ..
}
```

Reply:
Tiny nit but I would prefer `DocumentOptions`. I don't mind a little abbreviation for variables and small stuff but for types I'd prefer to keep them fully expanded